tests/test_x509_ext.py

```diff
@@ -3526,3 +3529,34 @@ def test_invalid_certificate_policies_data(self, backend):
         )
         with pytest.raises(ValueError):
             cert.extensions
+
+
+def _make_certbuilder(private_key):
+    name = x509.Name([x509.NameAttribute(NameOID.COMMON_NAME, u'example.org')])
+    builder = (
+        x509.CertificateBuilder()
+        .subject_name(name)
+        .issuer_name(name)
+        .public_key(private_key.public_key())
+        .serial_number(777)
+        .not_valid_before(datetime.datetime(1999, 1, 1))
+        .not_valid_after(datetime.datetime(2020, 1, 1))
+    )
+    return builder
+
+
+class TestCertificateBuilderExtensions(object):
+    @pytest.mark.requires_backend_interface(interface=RSABackend)
+    @pytest.mark.requires_backend_interface(interface=X509Backend)
+    def test_nameconstraints(self, backend):
+        permitted = [u'.example.org', u'.\xf5\xe4\xf6\xfc.example.com',
+                     u'x2603.example.net']
+        private_key = RSA_KEY_2048.private_key(backend)
+        builder = _make_certbuilder(private_key)
```

Comment:
`_make_certbuilder` is likely a good idea to reduce some duplication across our tests, but since it's only in use in this one place I'd prefer that you inlined it here. I'd be happy to review a PR that uses this technique to simplify the tests in the future though!